pkg/kubelet/dockertools/docker_manager.go

```diff
@@ -716,13 +720,23 @@ func (dm *DockerManager) runContainer(
 
 // setInfraContainerNetworkConfig sets the network configuration for the infra-container. We only set network configuration for infra-container, all
 // the user containers will share the same network namespace with infra-container.
-func setInfraContainerNetworkConfig(pod *api.Pod, netMode string, opts *kubecontainer.RunContainerOptions, dockerOpts *dockertypes.ContainerCreateConfig) {
+func (dm *DockerManager) setInfraContainerNetworkConfig(pod *api.Pod, netMode string, opts *kubecontainer.RunContainerOptions, dockerOpts *dockertypes.ContainerCreateConfig) {
 	exposedPorts, portBindings := makePortsAndBindings(opts.PortMappings)
 	dockerOpts.Config.ExposedPorts = exposedPorts
 	dockerOpts.HostConfig.PortBindings = dockernat.PortMap(portBindings)
 
+	result, err := dm.checkDockerAPIVersion(dockerV112APIVersion)
+	if err != nil {
+		glog.Errorf("Failed to check docker api version: %v", err)
+	}
+
+	hostNetPlusDNSEnabled := result >= 0
+
 	if netMode != namespaceModeHost {
 		dockerOpts.Config.Hostname = opts.Hostname
+	}
+
+	if netMode != namespaceModeHost || (netMode == namespaceModeHost && hostNetPlusDNSEnabled) {
```

Comment:
Hi Guys, docker 1.12 release date is 28th of July, so I think we need to decide smth. 

Let have a quick walk thru the history of the question first (all these can be drew out from docker issues discussion, just sum up it here):
1. initially this functionality was in docker, but had to be removed due to --net=host meant the host's /etc/resolv.conf was bind-mounted (https://github.com/docker/docker/issues/10384#issuecomment-72069550)
2. for now each containers recieves a private copy of resolv.conf
3. --net=host + dns settings already are available thru DOCKER_OPTS for a while
4. so the patch added fro docker 1.12 is about to enable usage of --net=host and --dns thru cli

Use cases:
1. if you don't want dns to be exposed to pod's resolv.conf there is already possibility for that: just set dnsPolicy:default

2. There are apps that use several networks, since it's not possible to assign multiple networks to container (Calico + Docker), the only way is left is to use
--net=host option and to edit resolv.conf on a host machines to get right resolv.conf for such pods. So it seems reasonable to use this enabled cli ability for
composing pod's resolv.conf in a desirable way.


And of course, as I wrote before we could do this support optional. I see here two options:
1. add kubelet option.
2. make dnsPolicy default be dependable on --net option: if --net=host is set then dnsPolicy is set to "default" by default.

WDYT?